src/rdf4cpp/rdf/parser/SerdIteratorImpl.ipp

```diff
@@ -0,0 +1,336 @@
+#include <rdf4cpp/rdf/parser/IStreamQuadIterator.hpp>
+
+#include <cassert>
+#include <cstddef>
+#include <deque>
+#include <istream>
+#include <utility>
+
+#if __has_include(<serd/serd.h>)
+#include <serd/serd.h>
+#else
+#include <serd-0/serd/serd.h>
+#endif
+
+
+#include <rdf4cpp/rdf/storage/util/robin-hood-hashing/robin_hood_hash.hpp>
+#include <rdf4cpp/rdf/Quad.hpp>
+
+namespace rdf4cpp::rdf::parser {
+
+struct IStreamAdaptor {
+    std::reference_wrapper<std::istream> istream;
+    uint32_t line_off = 0;
+    uint32_t col_off = 0;
+
+    template<bool track_position>
+    static size_t read(void *buf, size_t elem_size, size_t count, void *voided_self) {
+        assert(elem_size == 1);
+        assert(count == 1);
+
+        auto *self = reinterpret_cast<IStreamAdaptor *>(voided_self);
+        self->istream.get().read(reinterpret_cast<char *>(buf), static_cast<std::streamsize>(count));
+
+        if constexpr (track_position) {
+            if (static_cast<char const *>(buf)[0] == '\n') {
+                self->line_off += 1;
+                self->col_off = 0;
+            }
+        }
+
+        return self->istream.get().gcount();
+    }
+
+    static int is_ok(void *voided_self) {
+        auto *self = reinterpret_cast<IStreamAdaptor *>(voided_self);
+        return self->istream.get() ? 0 : 1;
+    }
+};
+
+struct IStreamQuadIterator::Impl {
+private:
```

Comment:
Separate source and header + add noexcepts

Reply:
We should not export the headers of serd what we currently would need to do. Right now, we would need to move serd in the CMake file at target_link_libraries to the PUBLIC section and remove the private mark in the conanfile.

There basically two options here:
a. move all serd imports to cpp files
b. use  a private header for IStreamQuadIteratorSerdBackend ( see https://github.com/dice-group/tentris-private/blob/develop/libs/triple-store/private-include/dice/triple-store/SerdLoad.hpp for an example ). 

Please check what's feasible here. I assume that only option b. will work.

Reply:
It's already not exposing serd. Notice that the impl part is in an .ipp file (not sure if it's the correct extension) that is only included in the .cpp file. Basically the file is not supposed to be included anywhere public

Reply:
But I can certainly move it into a "private-include" folder

Reply:
Ah, right. I overlooked that. Then I would reduce this comment to just add noexcepts ;)

Reply:
I have no experience with using ipp files. They should be omitted from the installed library because we filter for *.hpp and *.h. I would expect that they are visible if you pull in the library with FetchContent.